napari/_vispy/vispy_base_layer.py

```diff
@@ -192,10 +192,106 @@ def _reset_base(self):
         self._on_scale_change()
         self._on_translate_change()
 
+    def find_coordinates_of_canvas_corners(self, rounding_factor=None):
+        """Find location of the corners of canvas in data coordinates.
+
+        This method should only be used during 2D image viewing. The result
+        depends on the current pan and zoom position. Note that the returned
+        coordinates have been clipped to be inside the data to reflect the
+        actual amount of data that would be needed to cover canvas.
+
+        Parameters
+        ----------
+        rounding_factor : int, optional
+            Use a rounding factor to prevent repeated triggering of requesting
+            of new pyramid tiles for small camera movements. The rounding
+            factor is in pixel until of the currently viewed tile. Floor and
+            ceil rounding operations are used appropriately to ensure that the
+            rounded requested field always exceeds the needed field to fill the
+            canvas. There is a tradeoff between larger rounding factors causing
+            less frequent updates and tile fetches, but bigger tiles. A factor
+            of around 50 gives good results empically.
+
+        Returns
+        ----------
```

Comment:
could you make this dash as long as returns such that it looks like an underline?

Reply:
done, good catch! thanks